packages/next/src/client/components/app-router.tsx

```diff
@@ -424,7 +424,7 @@ function Router({
 
   const head = useMemo(() => {
     return findHeadInCache(cache, tree[1])
-  }, [cache, tree])
+  }, [cache, tree[1]])
```

Comment:
The current code is fine as-is, there won't be needless re-executions in this case, and this avoids accessing the property on every render.